### ISSUE TITLE
High-Performance Large-Scale Image Recognition Without Normalization

### ISSUE THREAD

**Issue by dltmddbs100** [dltmddbs100/Paper-log#13]
## Main Idea
**Efficient large batch training & High perfromance without normalization**
Paper: <https://arxiv.org/abs/2102.06171>

## Abstract
Batch normalization is a key component of most image classification models, but it has many undesirable properties stemming from its dependence on the batch size and interactions between examples. Although recent work has succeeded in training deep ResNets without normalization layers, these models do not match the test accuracies of the best batch-normalized networks, and are often unstable for large learning rates or strong data augmentations. In this work, we develop an adaptive gradient clipping technique which overcomes these instabilities, and design a significantly improved class of Normalizer-Free
ResNets. Our smaller models match the test accuracy of an EfficientNet-B7 on ImageNet while being up to 8.7× faster to train, and our largest models attain a new state-of-the-art top-1 accuracy of 86.5%. In addition, Normalizer-Free models attain significantly better performance than their batch-normalized counterparts when finetuning on ImageNet after large-scale pre-training on a dataset of 300 million labeled images, with our best models obtaining an accuracy of 89.2%.

## Contribution
+ Bacth normalization의 사용에 따른 단점들이 존재, 별도의 normalization없는 안정적인 모델을 구축
+ Normalizer-Free ResNets(NFNets) 구조를 기반으로 큰 batch size에도 안정적으로 학습하며 높은 성능과 속도를 갖는 방법을 고안 => Adaptive Gradient Clipping(AGC)
+ AGC는 특정 임계값을 설정해 모델 학습과정에서 gradient descent 과정에서 gradient가 임계값을 초과하지 않도록해 모델의 안정적인 학습을 돕는 방법
+ AGC는 gradient norm이 아닌 변화하는 weight에 대해 gradient의 변화도를 지표로 사용해 clipping
+ Ablation study of AGC 
 1. 최적의 λ는 optimizer, learning rate, batch size에 의존하며, batch size가 클수록 작은 λ를 채택하는 것이 효과적
 2. 작은 batch size에서는 AGC의 효과가 감소
 3. Final linear layer에는 AGC를 적용하지 않음
 4. 첫번째 convolution layer는 AGC가 없어도 안정적으로 학습이 가능한 경우가 많음
+ NFNet-F5 모델을 사용해 4096의 큰 batch size, 큰 learning rate으로 ImageNet sota달성
+ NFNet-F1은 EffNet-B7과 유사한 성능, 약 8.7배 빠른 학습 속도

## Reviews
<https://deepkerry.tistory.com/24>
